GitHubAPI/Program.cs

```diff
@@ -1,38 +1,70 @@
-﻿using Microsoft.Extensions.Configuration;
+﻿using GitHubAPI.Services;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using System;
 using System.IO;
-using System.Net.Http;
+using System.Threading.Tasks;
+using AutoMapper;
+using GitHubAPI.Model.JsonResponse;
+using Newtonsoft.Json;
 
 namespace GitHubAPI
 {
     class Program
     {
-        static void Main(string[] args)
+        static async Task Main(string[] args)
         {
             var configurationBuilder = new ConfigurationBuilder();
             BuildConfig(configurationBuilder);
 
             var host = Host.CreateDefaultBuilder()
-                .ConfigureServices((contect, services) =>
+                .ConfigureServices((context, services) =>
                 {
                     services.AddHttpClient(name: "GitHub", client =>
                       {
                           client.BaseAddress = new Uri("https://api.github.com");
                           client.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
                           client.DefaultRequestHeaders.Add("User-Agent", "HttpClientFactoryExample");
                       });
+                    services.AddTransient<IGitHubService, GitHubService>();
+                    services.AddAutoMapper(typeof(Program));
                 })
                 .Build();
+
+            var gitHubService = ActivatorUtilities.CreateInstance<GitHubService>(host.Services);

```

Comment:
Resolve that from DI container. Current implementation touches reflection and creates new instance of the class